doc/generate_cet.py

```diff
@@ -3,7 +3,121 @@
 import pathlib
 
 
+def provide_ignored():
```

Comment:
Please add a doc string to this and all other functions. It makes the code much easier to understand and maintain.